### PR TITLE
[2.0] make npx.load support empty .npz files

### PR DESCRIPTION
**Pull request by barry-jin**
## Description ##
Issue and bugfix are duplicate to https://github.com/numpy/numpy/pull/11522
cc @leezu @Zha0q1 @waytrue17 

### Essentials ###
- [x] PR's title starts with a category (e.g. [BUGFIX], [MODEL], [TUTORIAL], [FEATURE], [DOC], etc)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

File: src/c_api/c_api.cc [hunk 1]
```diff
@@ -1974,7 +1974,8 @@ int MXNDArrayLoad(const char* fname,
         << "Failed to read 32 bits from file.";
   }
 
-  if (magic == 0x04034b50 || magic == 0x504b0304) {  // zip file format; assumed to be npz
+  if (magic == 0x04034b50 || magic == 0x504b0304 || magic == 0x06054b50) {
```
Comment (leezu) [inline src/c_api/c_api.cc:1977]: You need to consider both big and little endian systems for the magic check here.

**Review by leezu — APPROVED** [review]

**Review by Zha0q1 — APPROVED** [review]
lgtm! thanks for the fix!

**Comment by Zha0q1** [comment]
@barry-jin is this pr ready to merge?

**Comment by barry-jin** [comment]
Yes

**Pull request merged**